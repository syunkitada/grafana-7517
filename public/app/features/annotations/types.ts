import { PanelData, DataFrame, AnnotationEvent, TimeRange } from '@grafana/data';
import { DashboardModel, PanelModel } from '../dashboard/state';

export interface AnnotationQueryOptions {
  dashboard: DashboardModel;
  panel: PanelModel;
  range: TimeRange;
}

export interface AnnotationQueryResponse {
  /**
   * All the data flattened to a single frame
   */
  frame?: DataFrame;

  /**
   * The processed annotation events
   */
  events?: AnnotationEvent[];

  /**
   * The original panel response
   */
  panelData?: PanelData;
}
