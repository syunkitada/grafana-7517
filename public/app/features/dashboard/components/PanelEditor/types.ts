import { PanelModel } from '../../state/PanelModel';

export enum DisplayMode {
  Fill = 0,
  Fit = 1,
  Exact = 2,
}

export const displayModes = [
  { value: DisplayMode.Fill, label: 'Fill', description: 'Use all avaliable space' },
  { value: DisplayMode.Fit, label: 'Fit', description: 'Fit in the space keeping ratio' },
  { value: DisplayMode.Exact, label: 'Exact', description: 'Same size as the dashboard' },
];

export enum EditorTab {
  Query = 'query',
  Alerts = 'alerts',
  Transform = 'xform',
}

export const allTabs = [
  { tab: EditorTab.Query, label: 'Query', show: (panel: PanelModel) => true },
  { tab: EditorTab.Alerts, label: 'Alerts', show: (panel: PanelModel) => true },
  { tab: EditorTab.Transform, label: 'Transform', show: (panel: PanelModel) => true },
];
