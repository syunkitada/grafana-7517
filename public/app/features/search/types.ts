export enum DashboardSearchItemType {
  DashDB = 'dash-db',
  DashHome = 'dash-home',
  DashFolder = 'dash-folder',
}

export interface DashboardSection {
  id: number;
  uid?: string;
  title: string;
  expanded?: boolean;
  url: string;
  icon: string;
  score: number;
  hideHeader?: boolean;
  checked?: boolean;
  items: DashboardSectionItem[];
  toggle?: (section: DashboardSection) => Promise<DashboardSection>;
  selected?: boolean;
  type: DashboardSearchItemType;
}

export interface DashboardSectionItem {
  checked?: boolean;
  folderId?: number;
  folderTitle?: string;
  folderUid?: string;
  folderUrl?: string;
  id: number;
  isStarred: boolean;
  selected?: boolean;
  tags: string[];
  title: string;
  type: DashboardSearchItemType;
  uid: string;
  uri: string;
  url: string;
}

export interface DashboardTag {
  term: string;
  count: number;
}

export interface DashboardQuery {
  query: string;
  mode: string;
  tag: string[];
  starred: boolean;
  skipRecent: boolean;
  skipStarred: boolean;
  folderIds: number[];
}

export interface SectionsState {
  sections: DashboardSection[];
  allChecked: boolean;
  dashboardTags: DashboardTag[];
}

export type ItemClickWithEvent = (item: DashboardSectionItem | DashboardSection, event: any) => void;

export type SearchAction = {
  type: string;
  payload?: any;
};

export interface OpenSearchParams {
  query?: string;
}
