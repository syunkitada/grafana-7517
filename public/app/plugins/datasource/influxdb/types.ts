import { DataQuery, DataSourceJsonData } from '@grafana/data';

export interface InfluxOptions extends DataSourceJsonData {
  timeInterval: string;
  httpMode: string;

  // Influx 2.0
  enableFlux?: boolean;
  organization?: string;
  defaultBucket?: string;
  maxSeries?: number;
}

export interface InfluxSecureJsonData {
  password?: string;
  token?: string;
}

export interface InfluxQueryPart {
  type: string;
  params?: string[];
  interval?: string;
}

export interface InfluxQueryTag {
  key: string;
  operator?: string;
  condition?: string;
  value: string;
}

export enum InfluxQueryType {
  Classic = 'Classic', // IFQL query builder
  InfluxQL = 'InfluxQL', // raw ifql
  Flux = 'Flux',
}

export interface InfluxQuery extends DataQuery {
  queryType?: InfluxQueryType;
  policy?: string;
  measurement?: string;
  resultFormat?: 'time_series' | 'table';
  orderByTime?: string;
  tags?: InfluxQueryTag[];
  groupBy?: InfluxQueryPart[];
  select?: InfluxQueryPart[][];
  limit?: string;
  slimit?: string;
  tz?: string;
  fill?: string;
  rawQuery?: boolean; // deprecated (use raw InfluxQL)
  query?: string;
}
