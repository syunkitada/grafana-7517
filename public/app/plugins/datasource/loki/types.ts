import { DataQuery } from '@grafana/ui/src/types';

export interface LokiQuery extends DataQuery {
  expr: string;
}

