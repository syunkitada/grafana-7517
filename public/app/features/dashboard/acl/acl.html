<div class="modal-body">
  <div class="modal-header">
    <h2 class="modal-header-title">
      <i class="fa fa-lock"></i>
      <span class="p-l-1">Permissions</span>
    </h2>

    <a class="modal-header-close" ng-click="ctrl.dismiss();">
      <i class="fa fa-remove"></i>
    </a>
  </div>

  <div class="modal-content">
    <table class="filter-table gf-form-group">
      <tr ng-repeat="acl in ctrl.items" ng-class="{'gf-form-disabled': acl.inherited}">
        <td style="width: 100%;">
          <i class="{{acl.icon}}"></i>
          <span ng-bind-html="acl.nameHtml"></span>
        </td>
        <td>
          <em class="muted no-wrap" ng-show="acl.inherited">Inherited from folder</em>
        </td>
        <td class="query-keyword">Can</td>
        <td>
          <div class="gf-form-select-wrapper">
            <select class="gf-form-input gf-size-auto" ng-model="acl.permission" ng-options="p.value as p.text for p in ctrl.permissionOptions" ng-change="ctrl.permissionChanged(acl)" ng-disabled="acl.inherited"></select>
          </div>
        </td>
        <td>
          <a class="btn btn-inverse btn-small" ng-click="ctrl.removeItem($index)" ng-hide="acl.inherited">
            <i class="fa fa-remove"></i>
          </a>
        </td>
      </tr>
      <tr ng-show="ctrl.aclItems.length === 0">
        <td colspan="4">
          <em>No permissions. Will only be accessible by admins.</em>
        </td>
      </tr>
    </table>

    <div class="gf-form-inline">
      <form name="addPermission" class="gf-form-group">
        <h6 class="muted">Add Permission For</h6>
        <div class="gf-form-inline">
          <div class="gf-form">
            <div class="gf-form-select-wrapper">
              <select class="gf-form-input gf-size-auto" ng-model="ctrl.newType" ng-options="p.value as p.text for p in ctrl.aclTypes"  ng-change="ctrl.typeChanged()"></select>
            </div>
          </div>
          <div class="gf-form" ng-show="ctrl.newType === 'User'">
            <user-picker user-picked="ctrl.userPicked($user)"></user-picker>
          </div>
          <div class="gf-form" ng-show="ctrl.newType === 'Group'">
            <user-group-picker user-group-picked="ctrl.groupPicked($group)"></user-group-picker>
          </div>
        </div>
      </form>
      <div class="gf-form width-17">
        <span ng-if="ctrl.error" class="text-error p-l-1">
          <i class="fa fa-warning"></i>
          {{ctrl.error}}
        </span>
      </div>
    </div>

    <div class="gf-form-button-row text-center">
      <button type="button" class="btn btn-danger" ng-disabled="!ctrl.canUpdate" ng-click="ctrl.update()">
        Update Permissions
      </button>
      <a class="btn-text" ng-click="ctrl.dismiss();">Close</a>
    </div>
  </div>
</div>

  <!-- <br> -->
  <!-- <br> -->
  <!-- <br> -->
  <!--  -->
  <!-- <div class="permissionlist"> -->
  <!--   <div class="permissionlist__section"> -->
  <!--     <div class="permissionlist__section&#45;header"> -->
  <!--       <h6>Permissions</h6> -->
  <!--     </div> -->
  <!--     <table class="filter&#45;table form&#45;inline"> -->
  <!--       <thead> -->
  <!--         <tr> -->
  <!--           <th style="width: 50px;"></th> -->
  <!--           <th>Name</th> -->
  <!--           <th style="width: 220px;">Permission</th> -->
  <!--           <th style="width: 120px"></th> -->
  <!--         </tr> -->
  <!--       </thead> -->
  <!--       <tbody> -->
  <!--         <tr ng&#45;repeat="permission in ctrl.userPermissions" class="permissionlist__item"> -->
  <!--           <td><i class="fa fa&#45;fw fa&#45;user"></i></td> -->
  <!--           <td>{{permission.userLogin}}</td> -->
  <!--           <td class="text&#45;right"> -->
  <!--             <a ng&#45;click="ctrl.removePermission(permission)" class="btn btn&#45;danger btn&#45;small"> -->
  <!--               <i class="fa fa&#45;remove"></i> -->
  <!--             </a> -->
  <!--           </td> -->
  <!--         </tr> -->
  <!--         <tr ng&#45;repeat="permission in ctrl.userGroupPermissions" class="permissionlist__item"> -->
  <!--           <td><i class="fa fa&#45;fw fa&#45;users"></i></td> -->
  <!--           <td>{{permission.userGroup}}</td> -->
  <!--           <td><select class="gf&#45;form&#45;input gf&#45;size&#45;auto" ng&#45;model="permission.permissions" ng&#45;options="p.value as p.text for p in ctrl.permissionTypeOptions" ng&#45;change="ctrl.updatePermission(permission)"></select></td> -->
  <!--           <td class="text&#45;right"> -->
  <!--             <a ng&#45;click="ctrl.removePermission(permission)" class="btn btn&#45;danger btn&#45;small"> -->
  <!--               <i class="fa fa&#45;remove"></i> -->
  <!--             </a> -->
  <!--           </td> -->
  <!--         </tr> -->
  <!--         <tr ng&#45;repeat="role in ctrl.roles" class="permissionlist__item"> -->
  <!--           <td></td> -->
  <!--           <td>{{role.name}}</td> -->
  <!--           <td><select class="gf&#45;form&#45;input gf&#45;size&#45;auto" ng&#45;model="role.permissions" ng&#45;options="p.value as p.text for p in ctrl.roleOptions" ng&#45;change="ctrl.updatePermission(role)"></select></td> -->
  <!--           <td class="text&#45;right"> -->
  <!--  -->
  <!--           </td> -->
  <!--         </tr> -->
  <!--       </tbody> -->
  <!--     </table> -->
  <!--   </div> -->
  <!--   </div> -->
  <!-- </div> -->
