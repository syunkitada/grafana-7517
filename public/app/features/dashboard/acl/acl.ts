///<reference path="../../../headers/common.d.ts" />

import coreModule from 'app/core/core_module';
import appEvents from 'app/core/app_events';
import _ from 'lodash';

export class AclCtrl {
  dashboard: any;
  userAcl: DashboardAcl[];
  groupAcl: DashboardAcl[];
  permissionTypeOptions = [
    {value: 1, text: 'View'},
    {value: 2, text: 'Edit'},
    {value: 4, text: 'Admin'}
  ];

  roleOptions = [
    {value: 0, text: 'No Access'},
    {value: 1, text: 'View'},
    {value: 2, text: 'Edit'},
    {value: 4, text: 'Admin'}
  ];

  roles = [];

  type = 'User Group';
  permission = 1;
  userId: number;
  userGroupId: number;

  /** @ngInject */
  constructor(private backendSrv, private dashboardSrv) {
    this.userAcl = [];
    this.groupAcl = [];
    this.dashboard = dashboardSrv.getCurrent();
    this.get(this.dashboard.id);
  }

  get(dashboardId: number) {
    return this.backendSrv.get(`/api/dashboards/id/${dashboardId}/acl`)
      .then(result => {
        this.userAcl = _.filter(result, p => { return p.userId > 0;});
        this.groupAcl = _.filter(result, p => { return p.userGroupId > 0;});
        this.roles = this.setRoles(result);
      });
  }

  setRoles(result: any) {
    return [
      {name: 'Viewer', permissions: 1},
      {name: 'Editor', permissions: 2},
      {name: 'Admin', permissions: 4}
    ];
  }

  addPermission() {
    if (this.type === 'User') {
      if (!this.userId) {
        return;
      }
      return this.addOrUpdateUserPermission(this.userId, this.permission).then(() => {
        this.userId = null;
        return this.get(this.dashboard.id);
      });
    } else {
      if (!this.userGroupId) {
        return;
      }

      return this.addOrUpdateUserGroupPermission(this.userGroupId, this.permission).then(() => {
        this.userGroupId = null;
        return this.get(this.dashboard.id);
      });
    }
  }

  addOrUpdateUserPermission(userId: number, permissions: number) {
    return this.backendSrv.post(`/api/dashboards/id/${this.dashboard.id}/acl`, {
      userId: userId,
      permissions: permissions
    });
  }

  addOrUpdateUserGroupPermission(userGroupId: number, permissions: number) {
    return this.backendSrv.post(`/api/dashboards/id/${this.dashboard.id}/acl`, {
      userGroupId: userGroupId,
      permissions: permissions
    });
  }

  updatePermission(permission: DashboardAcl) {
    if (permission.userId > 0) {
      return this.addOrUpdateUserPermission(permission.userId, permission.permissions);
    } else {
      if (!permission.userGroupId) {
        return;
      }
      return this.addOrUpdateUserGroupPermission(permission.userGroupId, permission.permissions);
    }
  }

  removePermission(permission: DashboardAcl) {
    return this.backendSrv.delete(`/api/dashboards/id/${permission.dashboardId}/acl/${permission.id}`).then(() => {
      return this.get(permission.dashboardId);
    });
  }
}

export function dashAclModal() {
  return {
    restrict: 'E',
    templateUrl: 'public/app/features/dashboard/acl/acl.html',
    controller: AclCtrl,
    bindToController: true,
    controllerAs: 'ctrl',
    scope: {
      dismiss: "&"
    }
  };
}

export interface FormModel {
  dashboardId: number;
  userId?: number;
  userGroupId?: number;
  PermissionType: number;
}

export interface DashboardAcl {
  id: number;
  orgId: number;
  dashboardId: number;
  created: Date;
  updated: Date;
  userId: number;
  userLogin: number;
  userEmail: string;
  userGroupId: number;
  userGroup: string;
  permissions: number;
  permissionName: string;
}

coreModule.directive('dashAclModal', dashAclModal);
