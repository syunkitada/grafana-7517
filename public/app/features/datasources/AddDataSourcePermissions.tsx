import React, { PureComponent } from 'react';
import { UserPicker } from 'app/core/components/Picker/UserPicker';
import { Team, TeamPicker } from 'app/core/components/Picker/TeamPicker';
import DescriptionPicker, { OptionWithDescription } from 'app/core/components/Picker/DescriptionPicker';
import { dataSourceAclLevels, AclTarget, DataSourcePermissionLevel } from 'app/types/acl';
import { User } from 'app/types';

export interface Props {
  onAddPermission: (state) => void;
  onCancel: () => void;
}

interface State {
  userId: number;
  teamId: number;
  type: AclTarget;
  permission: DataSourcePermissionLevel;
}

export class AddDataSourcePermissions extends PureComponent<Props, State> {
  cleanState = () => ({
    userId: 0,
    teamId: 0,
    type: AclTarget.Team,
    permission: DataSourcePermissionLevel.Query,
  });

  state = this.cleanState();

  isValid() {
    switch (this.state.type) {
      case AclTarget.Team:
        return this.state.teamId > 0;
      case AclTarget.User:
        return this.state.userId > 0;
    }
    return true;
  }

  onTeamSelected = (team: Team) => {
    this.setState({ teamId: team ? team.id : 0 });
  };

  onUserSelected = (user: User) => {
    this.setState({ userId: user ? user.id : 0 });
  };

  onPermissionChanged = (permission: OptionWithDescription) => {
    this.setState({ permission: permission.value });
  };

  onTypeChanged = event => {
    const type = event.target.value as AclTarget;

    this.setState({ type: type, userId: 0, teamId: 0 });
  };

  onSubmit = async event => {
    event.preventDefault();

    await this.props.onAddPermission(this.state);
    this.setState(this.cleanState());
  };

  render() {
    const { onCancel } = this.props;
    const { type, teamId, userId, permission } = this.state;

    const pickerClassName = 'width-20';
    const aclTargets = [{ value: AclTarget.Team, text: 'Team' }, { value: AclTarget.User, text: 'User' }];

    return (
      <div className="gf-form-inline cta-form">
        <button className="cta-form__close btn btn-transparent" onClick={onCancel}>
          <i className="fa fa-close" />
        </button>
        <form name="addPermission" onSubmit={this.onSubmit}>
          <h5>Add Permission For</h5>
          <div className="gf-form-inline">
            <div className="gf-form">
              <div className="gf-form-select-wrapper">
                <select className="gf-form-input gf-size-auto" value={type} onChange={this.onTypeChanged}>
                  {aclTargets.map((option, idx) => {
                    return (
                      <option key={idx} value={option.value}>
                        {option.text}
                      </option>
                    );
                  })}
                </select>
              </div>
            </div>
            {type === AclTarget.User && (
              <div className="gf-form">
                <UserPicker onSelected={this.onUserSelected} value={userId.toString()} className={pickerClassName} />
              </div>
            )}

            {type === AclTarget.Team && (
              <div className="gf-form">
                <TeamPicker onSelected={this.onTeamSelected} value={teamId.toString()} className={pickerClassName} />
              </div>
            )}
            <div className="gf-form">
              <DescriptionPicker
                optionsWithDesc={dataSourceAclLevels}
                onSelected={this.onPermissionChanged}
                value={permission}
                disabled={false}
                className={'gf-form-input--form-dropdown-right'}
              />
            </div>
            <div className="gf-form">
              <button data-save-permission className="btn btn-success" type="submit" disabled={!this.isValid()}>
                Save
              </button>
            </div>
          </div>
        </form>
      </div>
    );
  }
}

export default AddDataSourcePermissions;
