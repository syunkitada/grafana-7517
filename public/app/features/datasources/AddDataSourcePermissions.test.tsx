import React from 'react';
import { shallow } from 'enzyme';
import { AddDataSourcePermissions, Props } from './AddDataSourcePermissions';
import { AclTarget } from '../../types/acl';

const setup = () => {
  const props: Props = {
    onAddPermission: jest.fn(),
    onCancel: jest.fn(),
  };

  return shallow(<AddDataSourcePermissions {...props} />);
};

describe('Render', () => {
  it('should render component', () => {
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('should render user picker', () => {
    const wrapper = setup();

    wrapper.instance().setState({ type: AclTarget.User });

    expect(wrapper).toMatchSnapshot();
  });
});
