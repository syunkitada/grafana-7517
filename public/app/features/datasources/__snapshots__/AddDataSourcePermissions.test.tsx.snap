// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render component 1`] = `
<div
  className="gf-form-inline cta-form"
>
  <button
    className="cta-form__close btn btn-transparent"
    onClick={[MockFunction]}
  >
    <i
      className="fa fa-close"
    />
  </button>
  <form
    name="addPermission"
    onSubmit={[Function]}
  >
    <h5>
      Add Permission For
    </h5>
    <div
      className="gf-form-inline"
    >
      <div
        className="gf-form"
      >
        <select
          className="gf-form-input gf-size-auto"
          onChange={[Function]}
          value="Team"
        >
          <option
            key="0"
            value="Team"
          >
            Team
          </option>
          <option
            key="1"
            value="User"
          >
            User
          </option>
        </select>
      </div>
      <div
        className="gf-form"
      >
        <TeamPicker
          className="width-20"
          onSelected={[Function]}
        />
      </div>
      <div
        className="gf-form"
      >
        <DescriptionPicker
          className="gf-form-input--form-dropdown-right"
          disabled={false}
          onSelected={[Function]}
          optionsWithDesc={
            Array [
              Object {
                "description": "Can query data source.",
                "label": "Query",
                "value": 1,
              },
            ]
          }
          value={1}
        />
      </div>
      <div
        className="gf-form"
      >
        <button
          className="btn btn-success"
          data-save-permission={true}
          disabled={true}
          type="submit"
        >
          Save
        </button>
      </div>
    </div>
  </form>
</div>
`;

exports[`Render should render user picker 1`] = `
<div
  className="gf-form-inline cta-form"
>
  <button
    className="cta-form__close btn btn-transparent"
    onClick={[MockFunction]}
  >
    <i
      className="fa fa-close"
    />
  </button>
  <form
    name="addPermission"
    onSubmit={[Function]}
  >
    <h5>
      Add Permission For
    </h5>
    <div
      className="gf-form-inline"
    >
      <div
        className="gf-form"
      >
        <select
          className="gf-form-input gf-size-auto"
          onChange={[Function]}
          value="User"
        >
          <option
            key="0"
            value="Team"
          >
            Team
          </option>
          <option
            key="1"
            value="User"
          >
            User
          </option>
        </select>
      </div>
      <div
        className="gf-form"
      >
        <UserPicker
          className="width-20"
          onSelected={[Function]}
        />
      </div>
      <div
        className="gf-form"
      >
        <DescriptionPicker
          className="gf-form-input--form-dropdown-right"
          disabled={false}
          onSelected={[Function]}
          optionsWithDesc={
            Array [
              Object {
                "description": "Can query data source.",
                "label": "Query",
                "value": 1,
              },
            ]
          }
          value={1}
        />
      </div>
      <div
        className="gf-form"
      >
        <button
          className="btn btn-success"
          data-save-permission={true}
          disabled={true}
          type="submit"
        >
          Save
        </button>
      </div>
    </div>
  </form>
</div>
`;
